src/js/components/DataTable/DataTable.js

```diff
@@ -87,6 +87,7 @@ const DataTable = ({
   rowDetails,
   step = 50,
   verticalAlign,
+  allSelectedDisabled = false,
```

Comment:
```suggestion
  allowSelectAll = true,
```
Can we change the prop name here to `allowSelectAll`?